cmd/kube-proxy/app/server.go

```diff
@@ -522,21 +524,16 @@ with the apiserver API to configure the proxy.`,
 // ProxyServer represents all the parameters required to start the Kubernetes proxy server. All
 // fields are required.
 type ProxyServer struct {
-	Client                 clientset.Interface
-	Proxier                proxy.Provider
-	Broadcaster            events.EventBroadcaster
-	Recorder               events.EventRecorder
-	ConntrackConfiguration kubeproxyconfig.KubeProxyConntrackConfiguration
-	Conntracker            Conntracker // if nil, ignored
-	ProxyMode              kubeproxyconfig.ProxyMode
-	NodeRef                *v1.ObjectReference
-	MetricsBindAddress     string
-	BindAddressHardFail    bool
-	EnableProfiling        bool
-	OOMScoreAdj            *int32
-	ConfigSyncPeriod       time.Duration
-	HealthzServer          healthcheck.ProxierHealthUpdater
-	localDetectorMode      kubeproxyconfig.LocalMode
+	Config *kubeproxyconfig.KubeProxyConfiguration
```

Comment:
It seems it panics because this is nil https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-kubeadm-kinder-latest/1652574702679363584/artifacts/kinder-regular-control-plane-1/pods/kube-system_kube-proxy-74dfs_3a5b1b9b-a7e8-4716-9e99-6b2d80affb13/kube-proxy/5.log

Reply:
these are the kind job logs in this PR

```
2023-04-28T23:05:33.660333432Z stderr F I0428 23:05:33.660295       1 flags.go:64] FLAG: --detect-local-mode="ClusterCIDR"
2023-04-28T23:05:33.660349235Z stderr F I0428 23:05:33.660325       1 flags.go:64] FLAG: --feature-gates=""
2023-04-28T23:05:33.660545714Z stderr F I0428 23:05:33.660357       1 flags.go:64] FLAG: --healthz-bind-address="0.0.0.0:10256"
2023-04-28T23:05:33.660565048Z stderr F I0428 23:05:33.660414       1 flags.go:64] FLAG: --healthz-port="10256"
2023-04-28T23:05:33.660571874Z stderr F I0428 23:05:33.660451       1 flags.go:64] FLAG: --help="false"
2023-04-28T23:05:33.660603673Z stderr F I0428 23:05:33.660481       1 flags.go:64] FLAG: --hostname-override="kind-control-plane"
2023-04-28T23:05:33.660609867Z stderr F I0428 23:05:33.660509       1 flags.go:64] FLAG: --iptables-localhost-nodeports="true"
2023-04-28T23:05:33.660615225Z stderr F I0428 23:05:33.660535       1 flags.go:64] FLAG: --iptables-masquerade-bit="14"
2023-04-28T23:05:33.660627011Z stderr F I0428 23:05:33.660562       1 flags.go:64] FLAG: --iptables-min-sync-period="1s"
2023-04-28T23:05:33.660632551Z stderr F I0428 23:05:33.660589       1 flags.go:64] FLAG: --iptables-sync-period="30s"
2023-04-28T23:05:33.660642034Z stderr F I0428 23:05:33.660614       1 flags.go:64] FLAG: --ipvs-exclude-cidrs="[]"
2023-04-28T23:05:33.6606967Z stderr F I0428 23:05:33.660654       1 flags.go:64] FLAG: --ipvs-min-sync-period="0s"
2023-04-28T23:05:33.660740804Z stderr F I0428 23:05:33.660701       1 flags.go:64] FLAG: --ipvs-scheduler=""
2023-04-28T23:05:33.660787558Z stderr F I0428 23:05:33.660746       1 flags.go:64] FLAG: --ipvs-strict-arp="false"
2023-04-28T23:05:33.660800485Z stderr F I0428 23:05:33.660778       1 flags.go:64] FLAG: --ipvs-sync-period="30s"
2023-04-28T23:05:33.660888078Z stderr F I0428 23:05:33.660809       1 flags.go:64] FLAG: --ipvs-tcp-timeout="0s"
2023-04-28T23:05:33.660955756Z stderr F I0428 23:05:33.660896       1 flags.go:64] FLAG: --ipvs-tcpfin-timeout="0s"
2023-04-28T23:05:33.660967114Z stderr F I0428 23:05:33.660940       1 flags.go:64] FLAG: --ipvs-udp-timeout="0s"
2023-04-28T23:05:33.661005938Z stderr F I0428 23:05:33.660973       1 flags.go:64] FLAG: --kube-api-burst="10"
2023-04-28T23:05:33.661042625Z stderr F I0428 23:05:33.661015       1 flags.go:64] FLAG: --kube-api-content-type="application/vnd.kubernetes.protobuf"
2023-04-28T23:05:33.661086964Z stderr F I0428 23:05:33.661050       1 flags.go:64] FLAG: --kube-api-qps="5"
2023-04-28T23:05:33.661155334Z stderr F I0428 23:05:33.661110       1 flags.go:64] FLAG: --kubeconfig=""
2023-04-28T23:05:33.661220903Z stderr F I0428 23:05:33.661158       1 flags.go:64] FLAG: --log-flush-frequency="5s"
2023-04-28T23:05:33.661250448Z stderr F I0428 23:05:33.661198       1 flags.go:64] FLAG: --machine-id-file="/etc/machine-id,/var/lib/dbus/machine-id"
2023-04-28T23:05:33.661262325Z stderr F I0428 23:05:33.661232       1 flags.go:64] FLAG: --masquerade-all="false"
2023-04-28T23:05:33.661311276Z stderr F I0428 23:05:33.661265       1 flags.go:64] FLAG: --master=""
2023-04-28T23:05:33.661324961Z stderr F I0428 23:05:33.661299       1 flags.go:64] FLAG: --metrics-bind-address="127.0.0.1:10249"
2023-04-28T23:05:33.661362295Z stderr F I0428 23:05:33.661328       1 flags.go:64] FLAG: --metrics-port="10249"
2023-04-28T23:05:33.661418435Z stderr F I0428 23:05:33.661368       1 flags.go:64] FLAG: --nodeport-addresses="[]"
2023-04-28T23:05:33.661533357Z stderr F I0428 23:05:33.661413       1 flags.go:64] FLAG: --oom-score-adj="-999"
2023-04-28T23:05:33.661555104Z stderr F I0428 23:05:33.661523       1 flags.go:64] FLAG: --pod-bridge-interface=""
2023-04-28T23:05:33.661601989Z stderr F I0428 23:05:33.661560       1 flags.go:64] FLAG: --pod-interface-name-prefix=""
2023-04-28T23:05:33.661728992Z stderr F I0428 23:05:33.661601       1 flags.go:64] FLAG: --profiling="false"
2023-04-28T23:05:33.662118735Z stderr F I0428 23:05:33.662004       1 flags.go:64] FLAG: --proxy-mode="iptables"
2023-04-28T23:05:33.662136793Z stderr F I0428 23:05:33.662098       1 flags.go:64] FLAG: --proxy-port-range=""
2023-04-28T23:05:33.662240628Z stderr F I0428 23:05:33.662148       1 flags.go:64] FLAG: --show-hidden-metrics-for-version=""
2023-04-28T23:05:33.662250764Z stderr F I0428 23:05:33.662186       1 flags.go:64] FLAG: --v="4"
2023-04-28T23:05:33.662297878Z stderr F I0428 23:05:33.662253       1 flags.go:64] FLAG: --version="false"
2023-04-28T23:05:33.662343335Z stderr F I0428 23:05:33.662304       1 flags.go:64] FLAG: --vmodule=""
2023-04-28T23:05:33.66239284Z stderr F I0428 23:05:33.662343       1 flags.go:64] FLAG: --write-config-to=""
2023-04-28T23:05:33.664026234Z stderr F I0428 23:05:33.663804       1 feature_gate.go:249] feature gates: &{map[]}
2023-04-28T23:05:33.664203574Z stderr F I0428 23:05:33.664123       1 feature_gate.go:249] feature gates: &{map[]}
2023-04-28T23:05:33.678271685Z stderr F I0428 23:05:33.678128       1 node.go:141] Successfully retrieved node IP: 172.18.0.4
2023-04-28T23:05:33.678288459Z stderr F I0428 23:05:33.678165       1 server_others.go:119] "Detected node IP" address="172.18.0.4"
2023-04-28T23:05:33.718519435Z stderr F I0428 23:05:33.718366       1 server_others.go:190] "Using iptables Proxier"
2023-04-28T23:05:33.7186174Z stderr F I0428 23:05:33.718527       1 server_others.go:193] "kube-proxy running in dual-stack mode" ipFamily=IPv4
2023-04-28T23:05:33.718633164Z stderr F I0428 23:05:33.718604       1 server_others.go:194] "Creating dualStackProxier for iptables"
2023-04-28T23:05:33.718761171Z stderr F I0428 23:05:33.718688       1 server_others.go:487] "Unknown detect-local-mode" detectLocalMode=
2023-04-28T23:05:33.718847134Z stderr F I0428 23:05:33.718794       1 server_others.go:489] "Defaulting to no-op detect-local" detectLocalMode=""
``` 

the flag for the local-mode doesn't seem to be used @danwinship , it has to be something related to the detect-mode because it is the only thing that you changed, no?

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/117297/pull-kubernetes-e2e-kind/1652082938411487232/artifacts/kind-control-plane/pods/kube-system_kube-proxy-76s7t_6eac464e-3788-4250-a4df-588c8ed0d848/kube-proxy/0.log